op-deployer/pkg/deployer/opcm/opchain.go

```diff
@@ -105,39 +110,9 @@ type ReadImplementationAddressesOutput struct {
 	PreimageOracleSingleton      common.Address
 }
 
-type ReadImplementationAddressesScript struct {
-	Run func(input, output common.Address) error
-}
+type ReadImplementationAddressesScript script.DeployScriptWithOutput[ReadImplementationAddressesInput, ReadImplementationAddressesOutput]
 
-func ReadImplementationAddresses(host *script.Host, input ReadImplementationAddressesInput) (ReadImplementationAddressesOutput, error) {
-	var rio ReadImplementationAddressesOutput
-	inputAddr := host.NewScriptAddress()
-	outputAddr := host.NewScriptAddress()
-
-	cleanupInput, err := script.WithPrecompileAtAddress[*ReadImplementationAddressesInput](host, inputAddr, &input)
-	if err != nil {
-		return rio, fmt.Errorf("failed to insert ReadImplementationAddressesInput precompile: %w", err)
-	}
-	defer cleanupInput()
-	host.Label(inputAddr, "ReadImplementationAddressesInput")
-
-	cleanupOutput, err := script.WithPrecompileAtAddress[*ReadImplementationAddressesOutput](host, outputAddr, &rio,
-		script.WithFieldSetter[*ReadImplementationAddressesOutput])
-	if err != nil {
-		return rio, fmt.Errorf("failed to insert ReadImplementationAddressesOutput precompile: %w", err)
-	}
-	defer cleanupOutput()
-	host.Label(outputAddr, "ReadImplementationAddressesOutput")
-
-	deployScript, cleanupDeploy, err := script.WithScript[ReadImplementationAddressesScript](host, "ReadImplementationAddresses.s.sol", "ReadImplementationAddresses")
-	if err != nil {
-		return rio, fmt.Errorf("failed to load ReadImplementationAddresses script: %w", err)
-	}
-	defer cleanupDeploy()
-
-	if err := deployScript.Run(inputAddr, outputAddr); err != nil {
-		return rio, fmt.Errorf("failed to run ReadImplementationAddresses script: %w", err)
-	}
-
-	return rio, nil
+// NewDeployReadImplementationAddressesScript loads and validates the ReadImplementationAddresses script contract
```

Comment:
```suggestion
// NewReadImplementationAddressesScript loads and validates the ReadImplementationAddresses script contract
```

Also, unless we use `New` as convention throughout the code, I would use a clearer term such as `Load` - This reminds me of `NewNewFunctionv0-UPDATED` function names.

Reply:
Fixed via https://github.com/ethereum-optimism/optimism/pull/17542/commits/57ed1c161f457f60e84b337d4484707db3a79f55